### ISSUE TITLE
[Accessibility] [aria-*] attributes do not have valid values

### ISSUE THREAD

**Issue by mcolominas** [jackocnr/intl-tel-input#1369]
Assistive technologies, like screen readers, can't interpret ARIA attributes with invalid values. [Learn more](https://web.dev/aria-valid-attr-value/?utm_source=lighthouse&utm_medium=devtools).

```html
<div class="iti__selected-flag" role="combobox" aria-controls="iti-0__country-listbox" aria-owns="iti-0__country-listbox" aria-expanded="false" tabindex="0" aria-activedescendant="iti-0__item-es-preferred"></div>
```

**Comment by jackocnr** [comment]
Can you give a specific example of an ARIA attribute that you believe has an invalid value?

**Comment by mcolominas** [comment]
I just did some tests, from what I see, the aria that gives problems out of all the ones it has is "aria-activedescendant", this warning message appears when it is configured as "dropdownContainer: document.body".

When doing this, the list of countries is not loaded, therefore, it has no reference, the ideal in this case is that when in the configuration you have the dropdownContainer option, the aria-activedescendant are not loaded, and when the list appears/disappears, so does "aria-activedescendant".

Another error that I have found, the contrast ratio of ".iti__dial-code" is 2.54 and it should be a minimum of 4.5, for example use #767676, which is 4.54

**Comment by jackocnr** [comment]
Hmm this is weird - I don't understand how the attribute "aria-activedescendant" could have the value "dropdownContainer: document.body".

If you can find the minimal set of code that reproduces this behaviour, and put that in a CodePen, then I'd be happy to take a look. Here is [an example pen](https://codepen.io/jackocnr/pen/yehvE) with the plugin up and running - feel free to fork this if it helps.

**Comment by mcolominas** [comment]
When "dropdownContainer" is used, the list of countries is forced not to be present in the dom until it is requested, by doing this the id of "aria-activedescendant" is invalid because that list does not exist and it complains about that error, or at least, to me it is what makes me understand.

I have also realized that if the mobile version is used, the behavior is the same with or without "dropdownContainer", the "dropdownContainer" is only to force that the dom of the list is not loaded on the desktop

The code, it is something very basic.

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Phone</title>
    <link rel="stylesheet" href="intl-tel-input-master/build/css/intlTelInput.min.css">
    <script src="intl-tel-input-master/build/js/intlTelInput.min.js"></script>
</head>

<body>
    <h1>Phone</h1>
    <label for="phone">Phone</label>
    <input id="phone" type="tel">
    <script>
            const input = document.querySelector("#phone");
            window.intlTelInput(input, {
                dropdownContainer: document.body
            });
           // document.querySelector('.iti__selected-flag').removeAttribute('aria-activedescendant')
    </script>
</body>

</html>
```

**Comment by jackocnr** [comment]
Thank you , I understand the issue now. Yes we should put a guard around setting the "aria-activedescendant" attribute, and only do it if the country list is actually in the DOM e.g if dropdownContainer option is enabled (this is automatically enabled on mobile).

**Comment by jackocnr** [comment]
I've just pushed a fix for this, so now activedescendant is only set while the dropdown is open, which matches the behaviour here: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/

Thanks for raising this.
